### PR TITLE
fix(TDI-38724): Fix updating of custom records

### PR DESCRIPTION
**Pull request by iistomin-gl**
**What is the current behavior?** (You can also link to an open issue here)
- NetSuite output component is unable to update custom records

**What is the new behavior?**
- NetSuite output component correctly updates custom records

**Please check if the PR fulfills these requirements**

- [X] The commit message follows Talend standard
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [X] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes

**Does this PR introduce a breaking change?**

- [ ] Yes
- [X] No

**Other information**:
JIRA issue: https://jira.talendforge.org/browse/TDI-38724

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/318/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/318/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/321/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/321/)

File: components/components-netsuite/components-netsuite-runtime/src/main/java/org/talend/components/netsuite/NetSuiteDatasetRuntimeImpl.java [hunk 1]
```diff
@@ -91,16 +91,7 @@ public Schema getSchema(String typeName) {
...
-                @Override public int compare(FieldDesc o1, FieldDesc o2) {
-                    int result = Boolean.compare(o1.isKey(), o2.isKey());
-                    if (result != 0) {
-                        return result * -1;
-                    }
-                    result = o1.getName().compareTo(o2.getName());
-                    return result;
-                }
-            });
+            Collections.sort(fieldDescList, FieldDescComparator.INSTANCE);
```
Comment (dmytro-sylaiev) [inline components/components-netsuite/components-netsuite-runtime/src/main/java/org/talend/components/netsuite/NetSuiteDatasetRuntimeImpl.java]: Good move!

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/327/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/327/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/334/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/334/)

**Pull request merged**